core/src/main/scala/org/broadinstitute/dsde/rawls/billing/BpmBillingProjectCreator.scala

```diff
@@ -32,36 +42,61 @@ class BpmBillingProjectCreator(billingRepository: BillingRepository,
-          app.getTenantId == azureManagedAppCoordinates.tenantId
-      ) match {
-        case None =>
-          throw new ManagedAppNotFoundException(
-            ErrorReport(
-              StatusCodes.Forbidden,
-              s"Managed application not found [tenantId=${azureManagedAppCoordinates.tenantId}, subscriptionId=${azureManagedAppCoordinates.subscriptionId}, mrg_id=${azureManagedAppCoordinates.managedResourceGroupId}"
-            )
+      app.getSubscriptionId == azureManagedAppCoordinates.subscriptionId &&
+        app.getManagedResourceGroupId == azureManagedAppCoordinates.managedResourceGroupId &&
+        app.getTenantId == azureManagedAppCoordinates.tenantId
+    ) match {
+      case None =>
+        throw new ManagedAppNotFoundException(
+          RawlsErrorReport(
+            StatusCodes.Forbidden,
+            s"Managed application not found [tenantId=${azureManagedAppCoordinates.tenantId}, subscriptionId=${azureManagedAppCoordinates.subscriptionId}, mrg_id=${azureManagedAppCoordinates.managedResourceGroupId}"
           )
-        case Some(_) => Future.successful()
-      }
+        )
+      case Some(_) => Future.successful()
+    }
   }
 
   /**
    * Creates a billing profile with the given billing creation info and links the previously created billing project
    * with it
    */
   override def postCreationSteps(createProjectRequest: CreateRawlsV2BillingProjectFullRequest,
+                                 config: MultiCloudWorkspaceConfig,
                                  ctx: RawlsRequestContext
-  ): Future[Unit] = {
-    val profileModel = blocking {
-      billingProfileManagerDAO.createBillingProfile(createProjectRequest.projectName.value,
-        createProjectRequest.billingInfo,
-        ctx
-      )
+  ): Future[CreationStatus] =
+    try {
+      val profileModel = blocking {
+        billingProfileManagerDAO.createBillingProfile(createProjectRequest.projectName.value,
+                                                      createProjectRequest.billingInfo,
+                                                      ctx
+        )
+      }
+
+      val landingZoneResponse = blocking {
+        workspaceManagerDAO.createLandingZone(
+          config.azureConfig.get.landingZoneDefinition,
+          config.azureConfig.get.landingZoneVersion,
+          profileModel.getId,
+          ctx
+        )
+      }
+      val errorReport = Option(landingZoneResponse.getErrorReport)
+      errorReport match {
+        case Some(errorReport) =>
+          throw new LandingZoneCreationException(
+            RawlsErrorReport(StatusCode.int2StatusCode(errorReport.getStatusCode), errorReport.getMessage)
+          )
+        case None => ()
+      }
+      for {
+        _ <- billingRepository.storeLandingZoneCreationRecord(
+          UUID.fromString(landingZoneResponse.getJobReport.getId),
+          createProjectRequest.projectName.value
+        )
```

Comment:
We can't actually store the landing zone ID in the Rawls billing project yet because it isn't included the response until the landing zone has completed creation. See https://broadinstitute.slack.com/archives/C03JLE62QBZ/p1665590651845669